File: react-native-app/shared/models/SnitchTrigger.ts
import { LatLonPair } from "./CoordinateModels";
import RestaurantData from "./RestaurantData";

export default class SnitchTrigger {
    constructor(
        public created_at: number, 
        public originCoords: LatLonPair,
        public restaurantData: RestaurantData,
    ) {}
}

File: react-native-app/android/app/src/main/java/com/fitsnitchapp/SnitchTrigger.java
package com.fitsnitchapp;

public class SnitchTrigger {
    public long created_at;
    public LatLonPair originCoords;
    public Restaurant restaurantData;

    public SnitchTrigger(long created_at, LatLonPair coords, Restaurant restaurantData) {
        this.created_at = created_at;
        this.originCoords = coords;
        this.restaurantData = restaurantData;
    }
}